src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -1,33 +1,84 @@
-    Point ball_pos            = event.common.world_ptr->ball().position();
-    Rectangle field_lines     = event.common.world_ptr->field().fieldLines();
-    AutoChipOrKick auto_chick = {AutoChipOrKickMode::AUTOKICK,
-                                 WALL_KICKOFF_VELOCITY_M_PER_S};
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    pickoff_final_orientation =
+        calculateWallPickOffLocation(ball_pos, field_boundary,
+                                     MINIMUM_DISTANCE_FROM_WALL_FOR_ALIGN_METERS)
+            .first;
+    pickoff_point =
+        ball_pos - Vector::createFromAngle(pickoff_final_orientation).normalize(0.4);
+    align_wall_tactic->updateControlParams(
+        {.destination             = pickoff_point,
+         .final_orientation       = pickoff_final_orientation,
+         .final_speed             = 0.0,
+         .dribbler_mode           = TbotsProto::DribblerMode::OFF,
+         .ball_collision_type     = TbotsProto::BallCollisionType::AVOID,
+         .auto_chip_or_kick       = {AutoChipOrKickMode::OFF, 0},
+         .max_allowed_speed_mode  = TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+         .obstacle_avoidance_mode = TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE,
+         .target_spin_rev_per_s   = 0.0});
+    tactics_to_run[0].emplace_back(align_wall_tactic);
 
-    Angle kick_angle = calculateWallKickoffAngle(ball_pos, field_lines);
-    pivot_kick_tactic->updateControlParams(ball_pos, kick_angle, auto_chick);
-    tactics_to_run[0].emplace_back(pivot_kick_tactic);
+    event.common.set_tactics(tactics_to_run);
+}
+
+void BallPlacementPlayFSM::setPickOffDest(const BallPlacementPlayFSM::Update &event)
+{
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    std::pair<Angle, Point> location = calculateWallPickOffLocation(
+        ball_pos, field_boundary, MINIMUM_DISTANCE_FROM_WALL_FOR_ALIGN_METERS);
+
+    pickoff_final_orientation = location.first;
+    pickoff_destination =
+        location.second - Vector::createFromAngle(pickoff_final_orientation)
+                              .normalize(BACK_AWAY_FROM_WALL_M);
+}
+
+void BallPlacementPlayFSM::pickOffWall(const BallPlacementPlayFSM::Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{}};
+
+    // setup move skill tactics for robots away from ball placing robot
+    setupMoveSkillTactics(event);
+    tactics_to_run[0].insert(tactics_to_run[0].end(), move_skill_tactics.begin(),
+                             move_skill_tactics.end());
+
+    pickoff_wall_tactic->updateControlParams(
+        {.dribble_destination       = pickoff_destination,
+         .final_dribble_orientation = pickoff_final_orientation,
+         .excessive_dribbling_mode  = TbotsProto::ExcessiveDribblingMode::LOSE_BALL,
```

Comment:
should this be `ExcessiveDribblingMode::LOSE_BALL`? Shouldn't we allow excessive dribbling?